catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -224,20 +186,18 @@ private Object configure() {
   @SuppressWarnings(
       "squid:S1172" /* The code parameter is required in blueprint-cm-1.0.7. See https://issues.apache.org/jira/browse/ARIES-1436. */)
   public void destroy(int code) {
-    if (routeCollection == null) {
+    if (routeBuilder == null) {
```

Comment:
❗️ Access to `routeBuilder` is done from destroy and init threads. It should be under a lock protection

Reply:
@paouelle I have a question about that. Does blueprint not make any guarantees that destroy and init methods will be called exactly once?

Reply:
Synchronization is not just to ensure that code is called once or one at a time. It is also to make sure that common data is flushed or reloaded from memory in the processor's cache. If all you are doing is accessing/updating them from within the callbacks of blueprint, you should be fine. But if you are using separate threads of your own, then you got to put some synchronization blocks around access and updates to ensure the core's cache is properly flushed/reloaded.

Reply:
Since RouteBuilder won't be accessed too much outside startup, I just made it volatile.